pex/cli/commands/lock.py

```diff
@@ -76,11 +126,43 @@ def add_extra_arguments(
             name="create", help="Create a lock file.", func=cls._create
         ) as create_parser:
             cls._add_create_arguments(create_parser)
+        with subcommands.parser(
+            name="export", help="Export a Pex lock file in a different format.", func=cls._export
+        ) as export_parser:
+            cls._add_export_arguments(export_parser)
 
     def _create(self):
         # type: () -> Result
         requirement_configuration = requirement_options.configure(self.options)
         pip_configuration = resolver_options.create_pip_configuration(self.options)
+        network_configuration = pip_configuration.network_configuration
+
+        requirements = []  # type: List[Requirement]
+        local_projects = []  # type: List[LocalProjectRequirement]
+        for parsed_requirement in requirement_configuration.parse_requirements(
+            network_configuration
+        ):
+            if isinstance(parsed_requirement, LocalProjectRequirement):
+                local_projects.append(parsed_requirement)
+            else:
+                requirements.append(parsed_requirement.requirement)
+        if local_projects:
+            return Error(
+                "Cannot create a lock for local project requirements. Given {count}:\n"
+                "{projects}".format(
+                    count=len(local_projects),
+                    projects="\n".join(
+                        "{index}.) {project}".format(index=index, project=project.path)
+                        for index, project in enumerate(local_projects, start=1)
+                    ),
```

Comment:
I do too - sometimes apparently. Sometimes I do this and sometimes a numbered list. What I don't like is arbitrariness and fashion. What I should do is introduce a helper that creates a list using my whim style of the day that day and then just always use the utility instead of dithering with style back and forth every other week.